chain/cosmos/chain_node.go

```diff
@@ -861,3 +862,73 @@ func (tn *ChainNode) logger() *zap.Logger {
 		zap.String("test", tn.TestName),
 	)
 }
+
+// RegisterICA will attempt to register an interchain account on the counterparty chain.
+func (tn *ChainNode) RegisterICA(ctx context.Context, address, connectionID string) (string, error) {
+	command := []string{tn.Chain.Config().Bin, "tx", "intertx", "register",
+		"--from", address,
+		"--connection-id", connectionID,
+		"--chain-id", tn.Chain.Config().ChainID,
+		"--home", tn.NodeHome(),
+		"--node", fmt.Sprintf("tcp://%s:26657", tn.Name()),
+		"--keyring-backend", keyring.BackendTest,
+		"-y",
+	}
+
+	exitCode, stdout, stderr, err := tn.NodeJob(ctx, command)
+	if err != nil {
+		return "", dockerutil.HandleNodeJobError(exitCode, stdout, stderr, err)
+	}
+	output := IBCTransferTx{}
+	err = yaml.Unmarshal([]byte(stdout), &output)
+	if err != nil {
+		return "", err
+	}
+	return output.TxHash, nil
+}
+
+// QueryICA will query for an interchain account controlled by the specified address on the counterparty chain.
+func (tn *ChainNode) QueryICA(ctx context.Context, connectionID, address string) (string, error) {
+	command := []string{tn.Chain.Config().Bin, "query", "intertx", "interchainaccounts", connectionID, address,
+		"--chain-id", tn.Chain.Config().ChainID,
+		"--home", tn.NodeHome(),
+		"--node", fmt.Sprintf("tcp://%s:26657", tn.Name())}
+
+	exitCode, stdout, stderr, err := tn.NodeJob(ctx, command)
+	if err != nil {
+		return "", dockerutil.HandleNodeJobError(exitCode, stdout, stderr, err)
+	}
+
+	// at this point stdout should look like this:
+	// interchain_account_address: cosmos1p76n3mnanllea4d3av0v0e42tjj03cae06xq8fwn9at587rqp23qvxsv0j
+	// we split the string at the : and then just grab the address before returning.
+	parts := strings.SplitN(stdout, ":", 2)
+	return strings.TrimSpace(parts[1]), nil
+}
+
+// SendICABankTransfer builds a bank transfer message for a specified address and sends it to the specified
+// interchain account.
+func (tn *ChainNode) SendICABankTransfer(ctx context.Context, connectionID, fromAddr string, amount ibc.WalletAmount) error {
+	msg := fmt.Sprintf(`{
```

Comment:
I think producing this JSON is safe here with valid arguments, but I would feel better about it as 
```go
json.Marshal(map[string]interface{}{
  "@type": "/cosmos.bank.v1beta1.MsgSend",
  "from_address": fromAddr,
  "to_address", amount.Address,
  "amount": []map[string]interface{}{
    map[string]interface{}{
      "denom": amount.Denom,
      "amount": amount.Amount.String(), // Or fmt.Sprintf, whatever is appropriate.
    },
  },
})
```

Just to possibility of producing malformed JSON. If this were in wider use I might define local struct for it, but for a one-off, `map[string]interface{}` seems fine.

Reply:
Yeah I debated sticking this into a struct but I couldn't think of anywhere else we would use this unless we wanted to support generic bank messages. I'll make the changes so it's at least marshaling and we know its valid JSON.